packages/transaction-controller/src/TransactionController.ts

```diff
@@ -292,6 +292,7 @@ export class TransactionController extends BaseController<
       newTransactionMeta,
       'TransactionController#failTransaction - Add error message and set status to failed',
     );
+    this.onTransactionStatusChange(newTransactionMeta);
```

Comment:
Do we cover the scenario when the `PendingTransactionTracker` changes the status to `confirmed`?

Perhaps we can add this to `transaction-updated` listener in `addPendingTransactionTrackerListeners?